src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts

```diff
@@ -0,0 +1,377 @@
+import { ResultUpload } from '../result-client';
+import { DataField } from '../../domain/mi-export-data';
+import {
+  field,
+  formatManoeuvreFault,
+  formatMultipleManoeuvreFaults,
+  formatManoeuvreSerious,
+  optional,
+  optionalBoolean,
+  formatManoeuvreDangerous,
+  addIfSet,
+  getCompetencyComments,
+  formatQuestionFaultADI2,
+  formatQuestionCompleted,
+} from './data-mapper';
+import { formatGearboxCategory } from '../helpers/shared-formatters';
+import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
+import TestData = CatADI2UniqueTypes.TestData;
+
+/**
+ * Maps data specific to the ``ADI2`` test category.
+ *
+ * @param result The MES test result
+ * @returns The mapped MI Export data
+ * @throws MissingTestResultDataError If mandatory data missing from MES test result
+ */
+export const mapCatADI2Data = (result: ResultUpload): DataField[] => {
+  const t = result.testResult.testData;
```

Comment:
if we change this to be:
  const t = result.testResult.testData as TestData;
should't then have to:
t as TestData throughout, can just have t

Reply:
agreed. updated.